### PR TITLE
logging: integrate new IAM

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

**What type of PR is this?**
/kind cleanup
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

- Integreate new IAM
- Add more tests

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/api/logging/v1alpha2/types.go [hunk 1]
```diff
@@ -1,9 +1,103 @@
...
+
+const (
+	OperationStatistics = "statistics"
+	OperationHistogram  = "histogram"
+	OperationQuery      = "query"
+	OperationExport     = "export"
+
+	DefaultInterval = "15m"
+	DefaultSize     = 10
+	OrderAscending  = "acs"
```
[the remark this answers is not in the document]
Comment (huanggze) [reply pkg/api/logging/v1alpha2/types.go]: done

**Comment by huanggze** [comment]
/assign @wansir 

Can you help to review the code related to tenant?

**Comment by benjaminhuo** [comment]
/assign @zryfish 
@wansir hongming, is guangzhe's change ok for you?

**Comment by benjaminhuo** [comment]
@huanggze there're conflicts that must be resolved

File: pkg/models/tenant/tenant.go [hunk 2]
```diff
@@ -407,6 +414,95 @@ func (t *tenantOperator) Events(user user.Info, queryParam *eventsv1alpha1.Query
...
+		WorkloadSearch:  stringutils.Split(query.WorkloadSearch, ","),
+		WorkloadFilter:  stringutils.Split(query.WorkloadFilter, ","),
+		PodSearch:       stringutils.Split(query.PodSearch, ","),
+		PodFilter:       stringutils.Split(query.PodFilter, ","),
+		ContainerSearch: stringutils.Split(query.ContainerSearch, ","),
+		ContainerFilter: stringutils.Split(query.ContainerFilter, ","),
+		LogSearch:       stringutils.Split(query.LogSearch, ","),
+		Starttime:       query.StartTime,
+		Endtime:         query.EndTime,
+	}
```
Comment (wansir) [inline pkg/models/tenant/tenant.go:444]: Need to filter out the namespace（or worksapce） which the user doesn't have rights to list logs.
https://github.com/kubesphere/kubesphere/blob/master/pkg/models/tenant/tenant.go#L364-L382

**Pull request merged**